pkg/storage/gcp/bucket_cloud.go

```diff
@@ -0,0 +1,79 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"github.com/gomods/athens/pkg/config"
+	s "github.com/gomods/athens/pkg/storage"
+	multierror "github.com/hashicorp/go-multierror"
+	"google.golang.org/api/iterator"
+)
+
+// gcpBucket implements the (./pkg/storage/gcp).Bucket interface
+type gcpBucket struct {
+	*storage.BucketHandle
+}
+
+func (b *gcpBucket) Delete(ctx context.Context, module, version string) error {
+	if exists := b.Exists(ctx, module, version); !exists {
+		return s.ErrVersionNotFound{Module: module, Version: version}
+	}
+	var errs error
+	if err := b.Object(config.PackageVersionedName(module, version, "mod")).Delete(ctx); err != nil {
+		errs = multierror.Append(errs, err)
+	}
+	if err := b.Object(config.PackageVersionedName(module, version, "info")).Delete(ctx); err != nil {
+		errs = multierror.Append(errs, err)
+	}
+	if err := b.Object(config.PackageVersionedName(module, version, "zip")).Delete(ctx); err != nil {
```

Comment:
I think `Delete` should delete only one file just like Write writes only one file. Also because it will not be tested.

Reply:
I agree, and the check for not found should be moved